README.md

```diff
@@ -47,9 +47,14 @@ variations.
 
 Each test case has the the following keys:
 - description: which will be used to name each generated test
+  - the description should not simply explain **what** each case is (that is redundant information)
+  - the description should explain **why** each case is there
+  - for example, what kinds of incorrect implementations might this case help us find?
 - 'variable names': one or more variable names with values which will be passed to the solution method
-- expected: the expected result (this would be -1 when we expect an exception)
-- msg: a nice message for the failing case
+- expected: the expected result
+  - if an error is expected, the value at `"expected"` should be an object containing exactly one property, `"error"`, whose value is a string.
```

Comment:
This is also a point of attention to me. In #336 I suggest to use `null` as the value to represent a missing value. Languages with support for optional types can then use their optional type instead of null.